src/storage/hummock_sdk/src/table_watermark.rs

```diff
@@ -91,23 +91,76 @@ impl TableWatermarksIndex {
     pub fn range_watermarks(
         &self,
         epoch: HummockEpoch,
-        table_key_range: &TableKeyRange,
+        key_range: &mut TableKeyRange,
```

Comment:
Might assert the `rewrite` key_range should be a subset of the input key_range?

Reply:
We cannot have this assert, because when the whole query key range will be filtered out by the table watermark, we return a `(Included(watermark_key), Exclueded(watermark_key))` to mark an empty key range, which is not a subset of the original key range.

The added unit test should have covered most branches in the method, and it should be correct now.